src/Stack/Sig/Cabal.hs

```diff
@@ -0,0 +1,45 @@
+{-# LANGUAGE CPP               #-}
+{-# LANGUAGE FlexibleContexts  #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RankNTypes        #-}
+{-# LANGUAGE ViewPatterns      #-}
+
+{-|
+Module      : Stack.Sig.Cabal
+Description : Cabal Functions
+Copyright   : (c) FPComplete.com, 2015
+License     : BSD3
+Maintainer  : Tim Dysinger <tim@fpcomplete.com>
+Stability   : experimental
+Portability : POSIX
+-}
+
+module Stack.Sig.Cabal (cabalFilePackageId) where
+
+#if __GLASGOW_HASKELL__ < 710
+import           Control.Applicative ((<$>), (<*>))
+#endif
+
+import           Control.Monad.Catch
+import           Control.Monad.IO.Class
+import           Control.Monad.Logger
+import           Control.Monad.Trans.Control
+import qualified Data.Version as V
+import qualified Distribution.Package as D
+import qualified Distribution.PackageDescription as D
+import qualified Distribution.PackageDescription.Parse as D
+import qualified Distribution.Verbosity as D
+import           Stack.Types
+
+-- | Extract the @PackageIdentifier@ given an exploded haskell package
+-- path.
+cabalFilePackageId
```

Comment:
I think this can be replaced by some functions in Stack.Package, which will (1) avoid code duplication, and (2) ensure that Cabal library workarounds (like BOM stripping) are centralized to one place.